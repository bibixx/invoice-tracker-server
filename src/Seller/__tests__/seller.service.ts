import sinon from "sinon";

import SellerModel from "../seller.model";
import SellerService from "../seller.service";
const Seller = <any>SellerModel;

describe("SellerService.getSellers", () => {
  beforeEach(() => {
    sinon.stub(Seller, "find");
  });

  afterEach(() => {
    Seller.find.restore();
  });

  it("should return all sellers for current user", async () => {
    const SS = SellerService(Seller);
    Seller.find.resolves({});

    await expect(
      SS.getSellers("53cb6b9b4f4ddef1ad47f943"),
    ).resolves.not.toThrow();
  });

  it("should throw error if userId is invalid", async () => {
    const SS = SellerService(Seller);
    Seller.find.resolves({});

    await expect(
      SS.getSellers("asd"),
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});

describe("SellerService.getSellerById", () => {
  beforeEach(() => {
    sinon.stub(Seller, "find");
  });

  afterEach(() => {
    Seller.find.restore();
  });

  it("should return all sellers for current user", async () => {
    const SS = SellerService(Seller);
    Seller.find.resolves({});

    await expect(
      SS.getSellerById("53cb6b9b4f4ddef1ad47f943", "53cb6b9b4f4ddef1ad47f943"),
    ).resolves.not.toThrow();
  });

  it("should throw error if sellerId is invalid", async () => {
    const SS = SellerService(Seller);
    Seller.find.resolves({});

    await expect(
      SS.getSellerById("53cb6b9b4f4ddef1ad47f943", "asd"),
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it("should throw error if userId is invalid", async () => {
    const SS = SellerService(Seller);
    Seller.find.resolves({});

    await expect(
      SS.getSellerById("asd", "53cb6b9b4f4ddef1ad47f943"),
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
