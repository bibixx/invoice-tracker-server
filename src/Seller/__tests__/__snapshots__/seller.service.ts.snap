// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerService.getSellerById should throw error if sellerId is invalid 1`] = `"sellerId must be a valid ObjectId"`;

exports[`SellerService.getSellerById should throw error if userId is invalid 1`] = `"userId must be a valid ObjectId"`;

exports[`SellerService.getSellers should throw error if userId is invalid 1`] = `"userId must be a valid ObjectId"`;
